types/index.d.ts

```diff
@@ -340,20 +352,59 @@ export interface CookiesOptions {
 }
 
 /** [Documentation](https://next-auth.js.org/configuration/events) */
-export type EventType =
-  | "signIn"
-  | "signOut"
-  | "createUser"
-  | "updateUser"
-  | "linkAccount"
-  | "session"
-  | "error"
+export type EventCallback<MessageType = any> = (
```

Comment:
Could we mark the default type for `MessageType` as [`unknown`](https://devblogs.microsoft.com/typescript/announcing-typescript-3-0-rc-2/#the-unknown-type) to be more type-safe? 💭

Reply:
This required tweaking how the callback object's types were constructed - now the type for the signOut and session events is passed as a generic and the JWT and Session constructions are type aliases.